onboarding_project/squad_pantry_app/models.py

```diff
@@ -150,3 +155,72 @@ class ConfigurationSettings(models.Model):
 
     def __str__(self):
         return self.constant
+
+
+class PerformanceMetrics(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True, unique=True)
+    average_throughput = models.IntegerField(editable=False)
+    average_turnaround_time = models.DurationField(editable=False)
+
+    @classmethod
+    def calculate_avg_performance_metrics(cls):
+        """
+        Calculate the performance metrics as per scheduled, from last entered entry in database to now.
+
+        """
+        if PerformanceMetrics.objects.all().count() < 1:
+            completed_orders_curr_range = Order.objects.filter(
+                status=Order.DELIVERED
+            )
+        else:
+            start_date_time = PerformanceMetrics.objects.latest('id').created_at
+            completed_orders_curr_range = Order.objects.filter(
+                status=Order.DELIVERED,
+                created_at__range=(start_date_time, timezone.now())
+            )
+        throughput = len(completed_orders_curr_range)
+
+        if throughput == 0:
+            return timedelta(seconds=0), 0
+
+        total_turnaround_time = 0
+        for obj in completed_orders_curr_range:
+            total_turnaround_time += (obj.closed_at - obj.created_at).total_seconds()
+
+        average_turnaround_time = total_turnaround_time/throughput
+        return timedelta(seconds=average_turnaround_time), throughput
+
+    @classmethod
+    def create_avg_performance_metrics(cls):
+        """
+        Insert the calculated metrics into Database
+
+        """
+        turnaround_time, throughput = PerformanceMetrics.calculate_avg_performance_metrics()
+        PerformanceMetrics.objects.create(average_throughput=throughput, average_turnaround_time=turnaround_time)
+
+    @classmethod
+    def get_metrics_data(cls, start_date, end_date):
+        """
+        Get average metrics for the given interval
+
+        Keyword arguments:
+        start_date - Metrics needed from this date
+        end_date - Metrics needed till this date
+        """
+        metrics = PerformanceMetrics.objects.filter(created_at__range=(start_date, end_date))
+        num_metric_records = len(metrics)
+        total_throughput = metrics.aggregate(Sum('average_throughput'))['average_throughput__sum']
+        total_turnaround_time = 0
+        for obj in metrics:
+            total_turnaround_time += obj.average_turnaround_time.total_seconds()
```

Comment:
Did you search if postgreSQL provides native ways to add/subtract days? Did you come across [this](https://wiki.postgresql.org/wiki/Working_with_Dates_and_Times_in_PostgreSQL) document?